### DISCUSSION TITLE
Is all packages necessary?

### DISCUSSION THREAD

**Discussion by TheWotan** [sidebase/sidebase#86] [Q&A] [answered]
I follow and practice with nuxt3 and your implementations and struggle to eliminate unused packages. 

For example, I really won't be forced to use Ant Design (previously) or NaiveUI (now) - so I need to eliminate them manually, and stuck with hard coupling with `@sidebase/core`. 

So can we have something like a wizard on install? And check what exactly I need to install.

**Comment by BracketJohn** [comment, marked as answer]
Hey @TheWotan!

Good q - we want to have as little coupling as possible and all modules should be removable! I like the idea of a wizard (: We already explored it a little bit and will explore it further soon, to make usage + setup easier.

For your concrete case: You can remove naive-ui without a problem: The only thing this will break is `TheWelcomeContainer`! So to remove it the only two things you should have to do are:
1. Remove lines 3-5 from the `app.vue` -> this is the welcome container
2. Remove `'@huntersofbook/naive-ui-nuxt'` from the `nuxt.config.ts`
3. Uninstall naive ui: `npm r @huntersofbook/naive-ui-nuxt`

Please try this out and let me know if there's any remaining issues! Because if so, it's really important for us to resolve them to ensure that modules stay de-coupled!

Edit: The modules are now decoupled - `sidebase/core` does not depend on Naive UI anymore

**Comment by TheWotan** [comment]
Hi @BracketJohn 

Thanks for your answer.

Ofc, I manage to remove NaiveUI. But when I just do it as you mention `npm i` did not work. As I discover - because of coupling in `sidebase/core` to NaiveUI (but maybe I'm wrong )
```
export default defineNuxtConfig({
  modules: ['@huntersofbook/naive-ui-nuxt']
})
```
  from `/my-sidebase-app/node_modules/@sidebase/core/nuxt.config.ts`

Anyway, big thanks for your work and for the Merino template. Just my advice to add the `linter` config to that template.

**Reply by BracketJohn** [reply]
Ah, I get it now! We'll look into it: Usually what is expected to happen is that everything still works, just with missing styling. Thanks for clearing this up.

And yes: The merino stack partly exists because of your feedback here ❤️ 

Just now release the last version (for tonight) 🎊  For everybody else: To manually configure what modules you want to use, just use the new sidebase CLI:
```
npm create sidebase@latest
```

This also works with `pnpm` and `yarn`. The CLI will guide you through a sidebase app creation process:
![image](https://user-images.githubusercontent.com/25911230/208319502-f18cbe5a-4b6a-4a23-9a40-009167b0ba90.png)


Please checkout the full docs if you want to learn more: https://sidebase.io/sidebase/welcome

@TheWotan mind setting this question to answered, if everything is clear now? (:

**Comment by BracketJohn** [comment]
@TheWotan we now fully removed the sub-dependency on Naive UI. So you can just remove it and everything should work just fine!
